### PR TITLE
Render Full Contact Form even when there are Validation Errors

### LINKED ISSUE #437: Contact form covered by footer
### URL

https://www.jupiterbroadcasting.com/contact/

### On what type of device(s) did you see this bug?

Phone

### On which browser(s) are you seeing the problem?

Firefox

### Other Browser

_No response_

### Summary

If you submit the contect form without filling out all the feilds it shows an error. On mobile, the causes the bottom on the form, including the submit button to be covered by the footer.

Screenshot_20220930-183539

### Contributing

- [x] I have read through the [Contributing Guidelines](https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/blob/main/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by CGBassPlayer**
Fixes #437

* Increase size of contact form `iframe` to show submit button even when there are validation errors in the form
* Implement basic tests to check if the submit button is within the bounds of the contact form `iframe`

contact form with errors

### CODE REVIEW

**Comment by gerbrent** [comment]
amazing @CGBassPlayer - you're on :fire lately!

**Review by elreydetoda — CHANGES_REQUESTED** [review]
This is awesome @CGBassPlayer!

You really did an amazing job figuring out how to determine if the button was off the screen or not! I thought you'd be able to do something like `.is_visible()` on a locator, but for some reason playwright isn't like that (it says it's visible in the iframe even when it's not :disappointed: ).

I don't know if that's a bug or something, but the way you're comparing the height of the iframe to the bottom of the button was ingenious! 

Just have a few suggestions to help tidy up the code a bit :slightly_smiling_face: 

Overall though, amazing job!!! :partying_face: :tada: :rocket:

**Comment by gerbrent** [comment]
nice work both of you! It's a joy just to watch this evolve..

**Comment by elreydetoda** [comment]
BTW, these are the screenshots getting captured:

image

image

**Comment by CGBassPlayer** [comment]
> I thought you'd be able to do something like `.is_visible()` on a locator, but for some reason playwright isn't like that (it says it's visible in the iframe even when it's not disappointed ).
>
> I don't know if that's a bug or something...

The `.is_visible()` is checking to see if an html element has hidden (ethier `display: none` or `visibility:hidden`) in any of its styling. So even if you can't see physically see it, it is still "visible" on the page
https://playwright.dev/python/docs/actionability#visible

> ...but the way you're comparing the height of the iframe to the bottom of the button was ingenious!
>
> Just have a few suggestions to help tidy up the code a bit :slightly_smiling_face:

Thank you so much @elreydetoda! This is actually my first ever test (my day job does not have a good way to handle anything other than manual testing :cry:) so I will look over what you suggested and update this PR

File: test/e2e/test_contact.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    page.goto("/contact")
+    contact_form: FrameLocator = page.frame_locator('#contact-frame')
+    submit_button: Locator = contact_form.locator("#saveForm")
+
+    submit_button.click()
+    page.wait_for_load_state("networkidle")
+
+    # Scroll to the top (Page is current about 1400px)
+    page.mouse.wheel(0.0, -1500.0)
+    time.sleep(1) # Give time to scroll to the top
```
[the remark this answers is not in the document]
Comment (CGBassPlayer) [reply test/e2e/test_contact.py:21]: I feel like there isn't a "*best way*" to handle this in playwright yet. I knew my solution would not work if we made the page more than 180px taller. My only concern with this solution is there is padding above the content on the page (room for the navbar), so I am not 100% sure that test would scroll completely to the top. Like would it accept the `h1` being behind the navbar as "in view"?

Given that the navbar padding is only 64 pixels, I think we will be fine for right now, but we should definitely keep an eye out for something that would force playwright to move back to the top of a page.

File: test/e2e/test_contact.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    frame_height: float = float(page.get_attribute(selector="#contact-frame", name="height")) + page.locator(selector="#contact-frame").bounding_box().get("y")
```
[the remark this answers is not in the document]
Comment (CGBassPlayer) [reply test/e2e/test_contact.py:23]: I agree that looks cleaner and easier to read

File: test/e2e/test_contact.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    button_location: dict = submit_button.bounding_box()
+    if button_location is None:
+        assert False, "Submit Button is hidden"
```
[the remark this answers is not in the document]
Comment (CGBassPlayer) [reply test/e2e/test_contact.py:27]: I will throw in a brief explanation

File: test/e2e/test_contact.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (elreydetoda) [reply test/e2e/test_contact.py]: So, in my testing locally it seemed like it goes all the way to the top even past the Contact header. You can launch playwright in debug mode (don't currently have this enabled in the container version (i.e. `make tests`) but if you've got it installed with pipenv then you can do this) by doing something like: `PWDEBUG=1 pytest --base-url http://localhost:1313` which'll give you the interface you see below.

[contact-test_scroll.webm](https://user-images.githubusercontent.com/10230166/194733610-2ff202cf-ebb2-4e82-9120-6c1d900fef27.webm)

**Comment by elreydetoda** [comment]
> > ...but the way you're comparing the height of the iframe to the bottom of the button was ingenious!
> > Just have a few suggestions to help tidy up the code a bit slightly_smiling_face
> 
> Thank you so much @elreydetoda! This is actually my first ever test (my day job does not have a good way to handle anything other than manual testing cry) so I will look over what you suggested and update this PR

That is so awesome! You did a really good job with it :grin: 

How was the experience overall? Still trying to get a feel on how people like using playwright and if it's approachable enough :sweat_smile:

**Comment by CGBassPlayer** [comment]
I think the tool is great. I really based about 80% of the tests on things you did with the home page.

The docs were a little bit confusing to follow at first. But after a few reads, I was able to get through them alright.

Overall though, I think people will have an easy time working with this in the future.

**Pull request merged**